EIPS/eip-draft-transaction-ttls.md

```diff
@@ -0,0 +1,61 @@
+
+Further, in cases where the network is busy, or a transaction is published with a low gas price, or with a nonce gap or other problem that prevents it from being mined immediately, the transaction may be executed at any time in the future. This can cause trouble for users, where an effect can take place much later than intended, with no clean way to cancel it.
+
+This proposal addresses both of these issues by allowing nodes to place a hard limit on how long they will retain and propagate transactions, limiting the impact of transaction pool bloat and allowing smarter eviction strategies, while also providing users with a way to submit transactions that self-destruct if not mined within a fixed time period.
+
+## Specification
+The following names are used throughout this specification:
+
+ - `valid_until_block` - The block number after which a transaction may no longer be included in the chain.
+ - `current_blocknumber` - The number of the current block.
+ - `transaction_ttl` - The maximum number of blocks in the future a transaction's `valid_until_block` field may be, in order to be accepted and relayed by a given node.
+ - `fork_blocknumber` - The block number at which this change becomes active.
+
+Transactions are RLP-encoded lists described in the whitepaper as containing the 9 elements `(T_n, T_p, T_g, T_t, T_v, T_d, T_w, T_r, T_s)`, or `(T_n, T_p, T_g, T_t, T_v, T_i, T_w, T_r, T_s)` for contract-creation transactions. This EIP adds the field `T_b` as a tenth element to both tuples. `T_b` is defined as the latest block number at which this transaction may be included in the chain, and is hereafter referred to equivalently as `valid_until_block`.
+
+Block validation rules are amended to state that a block is invalid if it contains a 10-element transaction whose `T_b` is less than the block's number.
+
+Nodes MUST accept both 9-element and valid 10-element transactions in mined blocks.
+
+Nodes should not relay transactions whose `valid_until_block` is in the past, as these transactions are un-mineable. Nodes should treat other nodes that relay such transactions to them as broken or malicious.
+
+Nodes are also encouraged to set a transaction TTL, and refuse to accept or relay transactions where `valid_until_block > current_blocknumber + transaction_ttl`. As an implementation guideline, we suggest a default transaction TTL of 6000 - approximately one day.
+
+In regards to transaction pool inclusion and relaying, nodes are further encouraged to treat 9-element transactions as having a `valid_until_block` value of `fork_blocknumber + transaction_ttl`.
+
+The standard JSON APIs accepting transaction parameters are amended to add a parameter `valid_until_block`. If not specified, it should default to `current_blocknumber + transaction_ttl`.
```

Comment:
I would guess a missing `valid_until_block` means a nine-element transaction without the new `valid_until_block` element.  That would be the backward compatible behavior.